docs/errors/E0451.md

```diff
@@ -0,0 +1,23 @@
+# E0451: misleading use of ',' operator in conditional statement
+
+Using comma operator within a conditional statement of a control block (if, for, while, switch) is misleading, because only the last expression will be used to evaluate the condition.
```

Comment:
Nitpick: Surround code with \`:

```suggestion
Using comma operator within a conditional statement of a control block (`if`, `for`, `while`, `switch`) is misleading, because only the last expression will be used to evaluate the condition.
```